src/components/expression/expression-panel.js

```diff
@@ -224,16 +218,66 @@ genotet.ExpressionPanel.prototype.updateGenes = function(gene) {
       text: gene
     };
   });
-
   var profile = this.container.find('#profile select').empty();
-  this.selectProfiles_ = profile.select2({
-    data: genes,
-    multiple: true
-  });
-  this.container.find('#profile .select2-container').css({
-    width: '100%'
-  });
-  this.updateGeneSelections();
+
+  if (!genes.length) {
+    return;
+  }
+
+  $.fn.select2.amd.require(['select2/data/array', 'select2/utils'],
```

Comment:
"I tried your approach and it doens't work"... Hmm, I don't want to write code :/, but anyway 

Add the below to the bottom of genotet.utils
```js
/** @private @const {number} */
genotet.utils.SELECT2_PAGE_SIZE_ = 20;

/**
 * @param {!Element} $element
 * @param {!Object} options
 * @constructor
 */
genotet.utils.CustomData = function($element, options) {
  /**
   * @type {{
   *   __super__: function(?, ?)
   * }}
   */(genotet.utils.CustomData).__super__.constructor.call(
    this, $element, options);
};

$.fn.select2.amd.require(['select2/data/array', 'select2/utils'],
  function(ArrayData, Utils) {
  var CustomData =
  /**
   * @type {{
   *   Extend: function(?, ?)
   * }}
   */(Utils).Extend(genotet.utils.CustomData, ArrayData);

  /**
   * The query of CustomData.
   * @param {!Object<{
   *   page: number
   * }>} params Parameters for the query of CustomData.
   * @param {function(?)} callback Callback function.
   */
  CustomData.prototype.query = function(params, callback) {
    var pageSize = genotet.utils.SELECT2_PAGE_SIZE_;
    if (!('page' in params)) {
      params.page = 1;
    }
    var genes = this.options.options.data;
    callback({
      results: genes.slice((params.page - 1) * pageSize,
        params.page * pageSize),
      pagination: {
        more: params.page * pageSize < genes.length
      }
    });
  };
});
```

Then edit the updateGenes so that it looks like
```js
/**
 * Updates the genes in the profile list.
 * Select2 will regenerate the selection list each time updated.
 * @param {!Array<string>} gene List of genes.
 */
genotet.ExpressionPanel.prototype.updateGenes = function(gene) {
  var genes = gene.map(function(gene) {
    return {
      id: gene,
      text: gene
    };
  });
  var profile = this.container.find('#profile select').empty();

  if (!genes.length) {
    return;
  }

  this.selectProfiles_ = profile.select2({
    data: genes,
    ajax: {},
    dataAdapter: genotet.utils.CustomData,
    multiple: true
  });

  this.container.find('#profile .select2-container').css({
    width: '100%'
  });
  this.updateGeneSelections();
};
```

And it should work the same as your version. CustomData is a service and require sets up the service so that it can be reused later. You just need to have a way to reference to the service, and pass your data into the service.